modules/042-kube-dns/templates/deployment.yaml

```diff
@@ -62,21 +62,7 @@ spec:
       # hardcoded, because kube-dns is a critical component and system-cluster-critical priority class exists by default
       priorityClassName: system-cluster-critical
       serviceAccountName: d8-kube-dns
-      # Policy #0: kube-dns must be able to work on every master and every specific node
-      tolerations:
-      - key: CriticalAddonsOnly
-        operator: Exists
-      - key: node-role.kubernetes.io/master
-      - key: node-role/system
-      - key: dedicated.deckhouse.io
-        operator: Equal
-        value: kube-dns
-      - key: dedicated.deckhouse.io
-        operator: Equal
-        value: system
-      - key: dedicated.deckhouse.io
-        operator: Equal
-        value: master
+      {{- include "helm_lib_tolerations" (tuple . "any-node-with-no-csi") | nindent 6 }}
```

Comment:
It should be ok because we have `requiredDuringScheduling` node affinity rules. There may be a problem with the `CriticalAddonsOnly` toleration because I'm pretty sure this was an MCS compatibility issue fix. However, it should be solved by setting custom toleration keys in the Deckhouse config.